webpack.config.js

```diff
@@ -170,3 +172,21 @@ module.exports = {
     topLevelAwait: true,
   },
 };
+
+module.exports = (env, argv) => {
+  // since STB  isolates the build, we need to compile the translations here for releases
```

Comment:
Under the hood `stb` uses `build` for perform `stb package` (which in turn calls `python -m build`).
`build` builds the package in an isolated env to create the source distribution (https://build.pypa.io/en/stable/)

That means, when compiling assets from `tox` (or even doing `npm run build` during development or in our CI) it would result in something like

```
# note I added a simple console print to demonstrate
Webpack is building the theme in my-local-dir/pydata-sphinx-theme/src/pydata_sphinx_theme/theme/pydata_sphinx_theme/static
Locale will be located in my-local-dir/pydata-sphinx-theme/src/pydata_sphinx_theme/locale
```

When calling via `stb package` or `python -m build` (which is what we do to build the dist) this is instead

```
Webpack is building the theme in /private/var/folders/6p/062gm41556s5p63x1555cjzw0000gn/T/build-via-sdist-rmc_7ti7/pydata_sphinx_theme-0.16.1.dev0/src/pydata_sphinx_theme/theme/pydata_sphinx_theme/static
Locale will be located in /private/var/folders/6p/062gm41556s5p63x1555cjzw0000gn/T/build-via-sdist-rmc_7ti7/pydata_sphinx_theme-0.16.1.dev0/src/pydata_sphinx_theme/locale
```

so if you compile the translation files first with `tox run -e i18n-compile` then the compiled `.mo` will not be included in the distribution files by default (this isolation helps to avoid unwanted files and what not to be added to the distribution packages afterall)